File: android/live-update-provider-mock/src/main/kotlin/io/ionic/liveupdateprovidermock/LiveUpdateProviderMockPlugin.kt
package io.ionic.liveupdateprovidermock

import android.content.Context
import com.getcapacitor.Plugin
import com.getcapacitor.annotation.CapacitorPlugin
import io.ionic.liveupdateprovider.ProviderManager
import io.ionic.liveupdateprovider.LiveUpdateProvider

@CapacitorPlugin(name = "LiveUpdateProviderMock")
class LiveUpdateProviderMockPlugin : Plugin(), LiveUpdateProvider {
    override fun createManager(context: Context, configuration: Map<String, Any>): ProviderManager {
        val providerConfig = MockProviderConfig.from(configuration)

        return MockProviderManager(
            context = context,
            config = providerConfig
        )
    }
}

File: ios/Sources/LiveUpdateProviderMock/LiveUpdateProviderMockPlugin.swift
import Capacitor
import LiveUpdateProvider

@objc(LiveUpdateProviderMockPlugin)
class LiveUpdateProviderMockPlugin: CAPPlugin, CAPBridgedPlugin, LiveUpdateProvider {
    public let identifier = "LiveUpdateProviderMockPlugin"
    public let jsName = "LiveUpdateProviderMock"
    public let pluginMethods: [CAPPluginMethod] = []

    public func createManager(configuration: [String: Any]) throws -> any ProviderManager {
        let config = try MockProviderConfig(configuration)
        return MockProviderManager(config: config)
    }
}
